src/methods/join.js

```diff
@@ -43,7 +43,8 @@ function joinGenerators(
   generator1: Generator,
   generator2: Generator
 ): Generator {
-  let result = new Generator(realm, "joined");
+  // TODO #2222: Check if `realm.pathConditions` is correct here.
```

Comment:
It is correct. The path conditions are restored when evaluateNodeForEffects returns, so the path would be whatever it was when the fork (that is now being joined) has occurred.